doc/development-process.md

```diff
@@ -0,0 +1,160 @@
+# Development Process
+
+Depending on your area of focus we're providing guidance of each kind of development setup you might want.
+
+Step one, you **must** have the following things installed:
+
+* [go](https://go.dev) v1.17 -- Primary development language.
+* [docker](https://www.docker.com/) -- Used for generating containers & testing kubernetes set-ups.
+* [helm](https://helm.sh/docs/intro/install/) v3. -- Package manager for kubernetes
+* [kind](https://kind.sigs.k8s.io/docs/user/quick-start#installation) -- Run kubernetes clusters in docker for testing.
+* [flux](https://fluxcd.io/docs/get-started/) -- Continuous delivery system for kubernetes that weave-gitops enriches.
+* [tilt](https://tilt.dev/) -- Automatically build and deploy to a local cluster.
+
+Now you have those things installed, lets go!
+
+## Quick start development setup
+
+Help me get going as fast as possible!
+
+Create a kind cluster. We'll do this so we can have a local cluster on our machine
+
+`$ kind create cluster --name=<some name>`
+
+Then we install flux on it.
+
+`$ flux install`
+
+Then we bring up tilt.
+
+`$ tilt up`
+
+Tilt will tell you the url, or hit the space bar and it'll automagically open a browser for you which will show your local cluster and the resources running. You should see 3 things, Tiltfile, dev-weave-gitops, uncategorised.
+
+Goto login with username: dev, password:dev at http://localhost:9001/sign_in
+
+When you make changes in the code with this setup, you have to wait for k8s to redeploy the updated pod, it isn't lightening instant fast but **it works** to get you going.
+
+Woop! It's working. It'll be empty cos you haven't created any flux objects. You can jump to [
```

Comment:
> Woop! It's working.

A screenshot of what is meant to be working may help here :) i.e. the empty UI

Reply:
will come later 👍🏻